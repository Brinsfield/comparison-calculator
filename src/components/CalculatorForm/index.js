import React, { Component } from 'react';
class CalculatorForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      brand: '',
      nnn: 'Heartland Beef Mash',
      protein: 0,
      fat: 0,
      caloric: 0
    };

    this.handleChangeBrand = this.handleChangeBrand.bind(this);
    this.handleChangeNNN = this.handleChangeNNN.bind(this);
    this.handleInputChange = this.handleInputChange.bind(this);
  }

  componentDidUpdate(prevProps, prevState) {
    if (this.state !== prevState) {
      if (
        this.state.brand &&
        this.state.protein &&
        this.state.fat &&
        this.state.caloric
      ) {
        this.props.completedForm(true);
        this.props.comparisonInfo(this.state);
      } else {
        this.props.completedForm(false);
      }
    }
  }

  handleChangeBrand(evt) {
    this.setState({
      brand: evt.target.value
    });
  }
  handleChangeNNN(evt) {
    this.setState({
      nnn: evt.target.value
    });
  }

  handleInputChange(evt) {
    this.setState({
      [evt.target.name]: +evt.target.value
    });
  }

  render() {
    let infoText = this.state.brand
      ? `${this.state.brand}'s nutrient info:`
      : `Dog food brand's nutrient info:`;

    return (
      <form>
        <label>Name of dog food brand</label>
        <select value={this.state.brand} onChange={this.handleChangeBrand}>
          <option value="">-Select a Brand-</option>
          <option value="Bonio">Bonio</option>
          <option value="Milk-Bone">Milk-Bone</option>
          <option value="Friskies">Friskies</option>
        </select>

        <label>Compare with NomNomNow</label>
        <select value={this.state.nnn} onChange={this.handleChangeNNN}>
          <option value="Porkalicious Potluck">Porkalicious Potluck</option>
          <option defaultValue="Heartland Beef Mash">
            Heartland Beef Mash
          </option>
          <option value="Chicken Chow-Wow">Chicken Chow-Wow</option>
          <option value="Tasty Turkey Fare">Tasty Turkey Fare</option>
        </select>

        <div>{infoText}</div>

        <label>
          % protein (min.)
          <input
            type="number"
            name="protein"
            onChange={this.handleInputChange}
          />
        </label>

        <label>
          % fat (min.)
          <input type="number" name="fat" onChange={this.handleInputChange} />
        </label>

        <label>
          Caloric density (kcal/kg)
          <input
            type="number"
            name="caloric"
            onChange={this.handleInputChange}
          />
        </label>
      </form>
    );
  }
}

export default CalculatorForm;
