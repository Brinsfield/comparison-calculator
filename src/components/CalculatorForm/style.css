.calculator-form {
  text-align: left;
}

.food-brand,
.inputs {
  display: flex;
  padding-bottom: 10px;
}

.other-food-brands,
.nnn-food-brand {
  width: 100%;
  padding-bottom: 10px;
}

.other-food-brands {
  padding-right: 10px;
}

.label {
  display: block;
  font-weight: 400;
  padding-bottom: 5px;
}

select {
  background: url('../../assets/icon_chevron.png') no-repeat 96% center;
  background-color: #fff;
  width: 100%;
  background-size: 10px;
  border: 1px solid #eee;
  border-radius: 2px;

  font-size: 1em;
  padding: 10px 30px 10px 10px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.padding-bottom-small {
  padding-bottom: 10px;
}

.padding-right-small {
  padding-right: 10px;
}

.inputs div {
  width: 100%;
}

input {
  background-color: #fff;
  border: 1px solid #eee;

  font-size: 1em;
  padding: 10px 0;
  width: 100%;
}
