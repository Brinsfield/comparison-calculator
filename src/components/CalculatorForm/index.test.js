import React from 'react';
import { shallow } from 'enzyme'; // how to mount the component
import toJson from 'enzyme-to-json';
import CalculatorForm from '.';

describe('<CalculatorForm />', () => {
  let wrapper;
  it('renders', () => {
    // smoke test!!!
    wrapper = shallow(<CalculatorForm />);
  });
  it('matches snapshot', () => {
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });
});
