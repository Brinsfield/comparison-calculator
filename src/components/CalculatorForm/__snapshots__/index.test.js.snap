// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<form
  className="calculator-form"
>
  <div
    className="food-brands padding-bottom-small"
  >
    <div
      className="other-food-brands padding-bottom-small"
    >
      <label
        className="label"
      >
        Name of dog food brand
      </label>
      <select
        onChange={[Function]}
        value=""
      >
        <option
          value=""
        >
          - Select a Brand -
        </option>
        <option
          value="Bonio"
        >
          Bonio
        </option>
        <option
          value="Milk-Bone"
        >
          Milk-Bone
        </option>
        <option
          value="Friskies"
        >
          Friskies
        </option>
      </select>
    </div>
    <div
      className="nnn-food-brand padding-bottom-small"
    >
      <label
        className="label"
      >
        Compare with NomNomNow
      </label>
      <select
        onChange={[Function]}
        value="Heartland Beef Mash"
      >
        <option
          value="Porkalicious Potluck"
        >
          Porkalicious Potluck
        </option>
        <option
          defaultValue="Heartland Beef Mash"
        >
          Heartland Beef Mash
        </option>
        <option
          value="Chicken Chow-Wow"
        >
          Chicken Chow-Wow
        </option>
        <option
          value="Tasty Turkey Fare"
        >
          Tasty Turkey Fare
        </option>
      </select>
    </div>
  </div>
  <div
    className="padding-bottom-small"
  >
    Dog food brand's nutrient info:
  </div>
  <div
    className="inputs"
  >
    <div
      className="padding-bottom-small padding-right-small ingredients"
    >
      <label
        className="label"
      >
        % protein (min.)
      </label>
      <input
        name="protein"
        onChange={[Function]}
        type="number"
      />
    </div>
    <div
      className="padding-bottom-small padding-right-small ingredients"
    >
      <label
        className="label"
      >
        % fat (min.)
      </label>
      <input
        name="fat"
        onChange={[Function]}
        type="number"
      />
    </div>
    <div
      className="padding-bottom-small ingredients"
    >
      <label
        className="label"
      >
        Caloric density (kcal/kg)
      </label>
      <input
        name="caloric"
        onChange={[Function]}
        type="number"
      />
    </div>
  </div>
</form>
`;

exports[`<CalculatorForm /> matches snapshot 1`] = `
<form
  className="calculator-form"
>
  <div
    className="food-brands padding-bottom-small"
  >
    <div
      className="other-food-brands padding-bottom-small"
    >
      <label
        className="label"
      >
        Name of dog food brand
      </label>
      <select
        onChange={[Function]}
        value=""
      >
        <option
          value=""
        >
          - Select a Brand -
        </option>
        <option
          value="Bonio"
        >
          Bonio
        </option>
        <option
          value="Milk-Bone"
        >
          Milk-Bone
        </option>
        <option
          value="Friskies"
        >
          Friskies
        </option>
      </select>
    </div>
    <div
      className="nnn-food-brand padding-bottom-small"
    >
      <label
        className="label"
      >
        Compare with NomNomNow
      </label>
      <select
        onChange={[Function]}
        value="Heartland Beef Mash"
      >
        <option
          value="Porkalicious Potluck"
        >
          Porkalicious Potluck
        </option>
        <option
          defaultValue="Heartland Beef Mash"
        >
          Heartland Beef Mash
        </option>
        <option
          value="Chicken Chow-Wow"
        >
          Chicken Chow-Wow
        </option>
        <option
          value="Tasty Turkey Fare"
        >
          Tasty Turkey Fare
        </option>
      </select>
    </div>
  </div>
  <div
    className="padding-bottom-small"
  >
    Dog food brand's nutrient info:
  </div>
  <div
    className="inputs"
  >
    <div
      className="padding-bottom-small padding-right-small ingredients"
    >
      <label
        className="label"
      >
        % protein (min.)
      </label>
      <input
        name="protein"
        onChange={[Function]}
        type="number"
      />
    </div>
    <div
      className="padding-bottom-small padding-right-small ingredients"
    >
      <label
        className="label"
      >
        % fat (min.)
      </label>
      <input
        name="fat"
        onChange={[Function]}
        type="number"
      />
    </div>
    <div
      className="padding-bottom-small ingredients"
    >
      <label
        className="label"
      >
        Caloric density (kcal/kg)
      </label>
      <input
        name="caloric"
        onChange={[Function]}
        type="number"
      />
    </div>
  </div>
</form>
`;
